proto/documents/v1alpha/web_publishing.proto

```diff
@@ -0,0 +1,330 @@
+
+    // Currently there can only be one owner of the site.
+    OWNER = 1;
+
+    // Editors are allowed to push content to the site.
+    EDITOR = 2;
+  }
+
+  // Mintter Account ID of the member.
+  string account_id = 1;
+
+  // Member's role on the site.
+  Role role = 2;
+}
+
+// Information about a Mintter Document published on a Mintter Site.
+// One Document ID can be published with different versions and/or
+// under different paths on a Mintter Web Site, so it can have
+// multiple of these records per site.
+message WebPublicationRecord {
+  string document_id = 1;
+
+  string version = 2;
+
+  string hostname = 3;
+
+  string path = 4;
+}
+
+// API exposed by the remote server that allows
+// publishing Mintter Documents to serve them on the Web.
+// The server implementing this API is responsible for authenticating
+// callers for its administrative RPCs.
+// This API is also exposed by the local application backend, which would act
+// as a proxy to the remote web server, handling the authentication with its device private key.
+// The local application backend would require a special metadata header specifying the hostname
+// of the Mintter Site to proxy the request to.
+//
+// TODO(burdiyan): Document name of the required metadata header.
+service WebSite {
+  // Creates a new invite token for registering a new member.
+  // Protected. Must require authentication.
+  rpc CreateInviteToken(CreateInviteTokenRequest) returns (InviteToken);
+
+  // Redeems a previously created invite token to register a new member.
+  rpc RedeemInviteToken(RedeemInviteTokenRequest) returns (RedeemInviteTokenResponse);
+
+  // Gets public-facing site information.
+  rpc GetSiteInfo(GetSiteInfoRequest) returns (SiteInfo);
+
+  // Updates public-facing site information. Doesn't support partial updates,
+  // hence all the fields must be provided.
+  // Protected. Must require authentication.
+  rpc UpdateSiteInfo(UpdateSiteInfoRequest) returns (SiteInfo);
+
+  // Lists registered members on the site.
+  // May be protected or public depending on the privacy policies of the web site.
+  rpc ListMembers(ListMembersRequest) returns (ListMembersResponse);
+
+  // Gets information about a specific member.
+  // May be protected or public depending on the privacy policies of the web site.
+  rpc GetMember(GetMemberRequest) returns (Member);
+
+  // Deletes an existing member.
+  // Protected. Must require authentication.
+  rpc DeleteMember(DeleteMemberRequest) returns (google.protobuf.Empty);
+
+  // Publishes a Mintter Document on the remote web server along with any supporting materials.
+  rpc PublishDocument(PublishDocumentRequest) returns (PublishDocumentResponse);
+
+  // Unpublishes a previously published Document.
+  rpc UnpublishDocument(UnpublishDocumentRequest) returns (UnpublishDocumentResponse);
+
+  // list all the published documents
+  rpc ListWebPublications(ListWebPublicationsRequest) returns (ListWebPublicationsResponse);
```

Comment:
Who is allowed to get the list of publications? @burdiyan @horacioh @ericvicenti
- Anyone with at least Editor permissions?
- Everyone?

Reply:
I think we talked about letting anyone listing publications. They can do it on the web site anyway.